app/tests/test_generic_api.py

```diff
@@ -402,3 +402,80 @@ def test_head_cache_header(self):
                 self.assertAlmostEqual((expires - now).total_seconds(),
                                        timedelta(seconds=3600).total_seconds(),
                                        delta=2)
+
+
+class ApiCORSHeaderTestCase(StacBaseTestCase):
+
+    @classmethod
+    @mock_s3_asset_file
+    def setUpTestData(cls):
+        cls.factory = Factory()
+        cls.collection = cls.factory.create_collection_sample(
+            name='collection-1',
+            db_create=True,
+        )
+        cls.item = cls.factory.create_item_sample(
+            collection=cls.collection.model,
+            name='item-1',
+            db_create=True,
+        )
+        cls.asset = cls.factory.create_asset_sample(
+            item=cls.item.model,
+            name='asset-1',
+            db_create=True,
+        )
+
+    def test_get_cors_header(self):
+        for endpoint in [
+            '',  # landing page
+            'conformance',
+            f'collections/{self.collection["name"]}',
+            f'collections/{self.collection["name"]}/items/{self.item["name"]}',
+            f'collections/{self.collection["name"]}/items/{self.item["name"]}'
+            f'/assets/{self.asset["name"]}'
+        ]:
+            with self.subTest(endpoint=endpoint):
+                response = self.client.get(f"/{STAC_BASE_V}/{endpoint}")
+                self.assertStatusCode(200, response)
+
+                self.assertTrue(
+                    response.has_header('Access-Control-Allow-Origin'),
+                    msg="CORS header allow-origin missing"
+                )
+                self.assertEqual(
+                    response['Access-Control-Allow-Origin'],
+                    '*',
+                    msg='Wrong CORS allow-origin value'
+                )
+
+                self.assertTrue(
+                    response.has_header('Access-Control-Allow-Methods'),
+                    msg="CORS header allow-methods missing"
+                )
+                self.assertEqual(
+                    response['Access-Control-Allow-Methods'],
+                    'GET,HEAD',
+                    msg='Wrong CORS allow-methods value'
+                )
+
+    def test_get_cors_header_search(self):
+        response = self.client.get(f"/{STAC_BASE_V}/search")
+        self.assertStatusCode(200, response)
+
+        self.assertTrue(
+            response.has_header('Access-Control-Allow-Origin'),
+            msg="CORS header allow-origin missing"
+        )
+        self.assertEqual(
+            response['Access-Control-Allow-Origin'], '*', msg='Wrong CORS allow-origin value'
+        )
+
+        self.assertTrue(
+            response.has_header('Access-Control-Allow-Methods'),
+            msg="CORS header allow-methods missing"
+        )
+        self.assertEqual(
+            response['Access-Control-Allow-Methods'],
+            'GET,HEAD,POST',
+            msg='Wrong CORS allow-methods value'
+        )
```

Comment:
Same as above the check for `Access-Control-Allow-Headers` is missing.

Reply:
added